app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java

```diff
@@ -91,14 +94,34 @@ public void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
 
         usernameEdit.addTextChangedListener(textWatcher);
+        usernameEdit.setOnFocusChangeListener((v, hasFocus) -> {
+            if (!hasFocus) {
+                hideKeyboard(v);
+            }
+        });
+
         passwordEdit.addTextChangedListener(textWatcher);
+        passwordEdit.setOnFocusChangeListener((v, hasFocus) -> {
+            if (!hasFocus) {
+                hideKeyboard(v);
+            }
+        });
+
         twoFactorEdit.addTextChangedListener(textWatcher);
         passwordEdit.setOnEditorActionListener(newLoginInputActionListener());
 
         loginButton.setOnClickListener(view -> performLogin());
         signupButton.setOnClickListener(view -> signUp());
     }
 
+
+    public void hideKeyboard(View view) {
+        Log.i("hide", "hideKeyboard: ");
```

Comment:
Please use Timber for logging. :)

Reply:
Can I simply remove the log statement instead @misaochan ?

Reply:
It does seem that this particular log statement isn't very useful, yes. I think removing it would be fine in this case.